docs/authoring/servertaskauthoring.md

```diff
@@ -0,0 +1,77 @@
+# Server Task Authoring
+
+In this document, we will talk about how you can author a new server tasks and how it is different from an agent task.
+
+You can take a look at this [document](https://docs.microsoft.com/en-us/vsts/extend/develop/add-build-task) to learn about how you can add a new agent task. Assuming that you are aware of the process for an adding an agent-based task, let us see what all is different in server base task.
+
+##### RunsOn:
+This property in task.json specifies where the task should run. The possible values for this property are as follows: -
+
+- Agent
+- DeploymentGroup
+- Server
+- ServerGate
+
+The default value of this property is Agent, but you can change it to Server to specify that the task should run on VSTS/TFS server.
+
+##### Execution section:
+This property defines which how the task should be executed. For agent-based tasks, you can specify use powershell/node handlers to run your custom powershell/node scripts but we don&#39;t support running them on VSTS/TFS server.
+
+Here are the possible actions you can do for server tasks.
+
+- **HttpRequest** :
+This enables you to invoke a Http end-point. It takes 3 inputs.
+
+    **Events:** This property gives control on task execution. &#39;Events&#39; property is optional. External systems respond with these events to indicate state changes using the TaskHttpEvents API.  You can configure timeouts for these events. You can use [ServerTaskHelper](https://github.com/Microsoft/vsts-rm-extensions/tree/master/ServerTaskHelper) to post the events and task execution logs to VSTS/TFS server. The possible values for this property are as follows: -
+    1.  TaskAssigned - External service raises to acknowledge that the ‘Execute’ event has been received (optional)
+    2.  TaskStarted - Raised when the external service starts processing
+    3.  TaskCompleted -Raised when the external system is done processing (indicates success/failure)
+    
+         *Example*:  In below example once server picks task for execution, if server not receives the &#39;TaskStarted&#39; event within 5 minutes, server cancels the task. If server not receives the &#39;TaskCompleted&#39; event within 7 minutes, server cancels the task.         
+   "Events": { "TaskStarted": {"Timeout": "00:05:00"}, "TaskCompleted": {"Timeout": "00:07:00"}}
+
+    **Execute:** Configures the HTTP payload and properties for the Execute event raised from VSTS to the cloud service.  'HttpRequest' handler supports following properties.
+    - EndpointId: - Specified endpointId details used while triggering below mentioned &#39;EndpointUrl&#39;. This input is optional
```

Comment:
Specifies the endpointId which should be used to generate the authentication header.